### PR TITLE
[CRS-25] add typescript build on js files

### PR DESCRIPTION
**Pull request by jaapbakker88**
# Submit a pull request

Typescript type checking on JS files based on JSDoc instead of converting all files to typescript. 
More info: https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#supported-jsdoc

## To check a file:
add `// @ts-check` to the top of the file

add the types to the comments: 
```
/**
 * Card - Simple Card Layout
 *
 * @example ./docs/Card.md
 * @type Card { import("../../types/index.d.ts").CardProps }
 */
```

ignore lines with:
```
// @ts-ignore
import giphyLogo from '../assets/Poweredby_100px-White_VertText.png';
```

run `tsc --strict`

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/Card.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+// @ts-ignore
 import giphyLogo from '../assets/Poweredby_100px-White_VertText.png';
 import { withTranslationContext } from '../context';
+
 /**
  * Card - Simple Card Layout
  *
  * @example ./docs/Card.md
- * @extends PureComponent
+ * @type Card { import("../../types/index.d.ts").CardProps }
```
Comment (vishalnarkhede) [inline src/components/Card.js:13]: but we can't typecheck the methods of component then, right?

**Comment by vishalnarkhede** [comment]
@jaapbakker88  with this approach, we can't typecheck the methods of component then, right?

File: src/components/Card.js [hunk 2]
```diff
@@ -26,6 +28,7 @@ class Card extends React.PureComponent {
     text: PropTypes.string,
   };
 
+  /** @type {(url: string) => any} Typescript syntax */
```
Comment (jaapbakker88) [inline src/components/Card.js:31]: @vishalnarkhede here you can add types to the internal methods

Comment (tbarbugli) [reply src/components/Card.js:31]: also good to keep in mind that classes will get killed in the hooks refactor ;)

**Comment by vishalnarkhede** [comment]
@jaapbakker88  I Love it!! 

Can we also create a wiki on this repo? And start writing short articles about dev setup. Typescript would be great place to start. Description of this PR is good enough for article, with some additions regarding missing types for external library (like vfile-message here)

**Pull request merged**